workflow/scripts/compare_pathways.py

```diff
@@ -32,62 +32,88 @@ def prepare(df):
+            ): pl.Float64
+        }
+    )
+    .with_columns(pl.min_horizontal("Combined FDR", "Combined FDR_y").alias("min fdr"))
+    .filter(pl.col("min fdr") <= 0.05)
     .rename(
         {
             "total perturbation accumulation": effect_x,
             "total perturbation accumulation_y": effect_y,
-            "fdr_min": "min fdr",
         }
     )
     .collect()
 )
 
-effects = combined.select(pl.col(effect_x, effect_y))
-min_value = effects.min().min_horizontal()[0]
-max_value = effects.max().max_horizontal()[0]
-combined = combined.with_columns(
-    abs(pl.col(effect_x) - pl.col(effect_y)).alias("difference")
+if not combined.is_empty():
+    combined = (
+        combined.with_columns(
+            abs(pl.col(effect_x) - pl.col(effect_y)).alias("difference")
+        )
+        .with_columns(
+            (-pl.col("min fdr").log(base=10) * pl.col("difference")).alias("pi_value")
+        )
+        .sort(pl.col("pi_value").abs(), descending=True)
+    )
+else:
+    combined = combined.with_columns(
+        [pl.lit(None).alias("difference"), pl.lit(None).alias("pi_value")]
+    )
+combined = combined.select(
+    [
+        "Name",
+        "min fdr",
+        effect_x,
+        effect_y,
+        "difference",
+        "pathway id",
+        "pi_value",
+    ]
 )
-combined_sorted = combined.sort("difference", descending=True)
-combined_pd = combined_sorted.select(
-    pl.col("Name", "min fdr", effect_x, effect_y, "difference", "pathway id")
-).to_pandas()
-combined_pd.to_csv(snakemake.output[0], sep="\t", index=False)
-
-# we cannot use vegafusion here because it makes the point selection impossible since
-# it prunes the required ext_gene column
-# alt.data_transformers.enable("vegafusion")
-alt.data_transformers.disable_max_rows()
+combined.write_csv(snakemake.output[0], separator="\t")
 
+df = combined.select(["min fdr", effect_x, effect_y]).to_pandas()
 
-alt.data_transformers.disable_max_rows()
-point_selector = alt.selection_point(fields=["Name"], empty=False)
+point_selector = alt.selection_single(fields=["term"], empty="all")
 
+alt.data_transformers.disable_max_rows()
 points = (
-    alt.Chart(combined_pd)
+    alt.Chart(df)
     .mark_circle(size=15, tooltip={"content": "data"})
     .encode(
         alt.X(
             effect_x,
+            title=label_x,
             scale=alt.Scale(type="symlog", nice=False),
-            axis=alt.Axis(grid=False),
+            axis=alt.Axis(grid=True),
         ),
         alt.Y(
             effect_y,
+            title=label_y,
             scale=alt.Scale(type="symlog", nice=False),
-            axis=alt.Axis(grid=False),
+            axis=alt.Axis(grid=True),
         ),
         alt.Color("min fdr", scale=alt.Scale(scheme="viridis")),
         opacity=alt.value(0.5),
     )
 )
 
+min_value = combined.select(pl.min(effect_x, effect_y)).min_horizontal()[0]
+max_value = combined.select(pl.max(effect_x, effect_y)).max_horizontal()[0]
```

Comment:
Is this intentionally different from this solution:
https://github.com/snakemake-workflows/rna-seq-kallisto-sleuth/pull/113/files#diff-237db913ebc1ad915aaaf23a78b5ee0bc0a47404c543b4fea06d35e24e76371eR100-R101